From b73fb49177dc50a7b7eefa45235b23f4cb8cdd4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pasi=20Sj=C3=B6holm?= <pasi.sjoholm@jollamobile.com>
Date: Wed, 11 Feb 2015 12:14:19 +0200
Subject: [PATCH 05/11] network: Fix eternal associating with IPv6-only

It was possible with IPv6-only networks to have eternal
associating variable set as true, especially
when ipv4 configuration method was OFF.
---
 src/network.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/network.c b/src/network.c
index db19cb9..0e979c9 100644
--- a/src/network.c
+++ b/src/network.c
@@ -342,6 +342,8 @@ static int manual_ipv6_set(struct connman_network *network,
 
 	connman_device_set_disconnected(network->device, false);
 
+	connman_network_set_associating(network, false);
+
 	network->connecting = false;
 
 	return 0;
@@ -386,8 +388,6 @@ static int dhcpv6_set_addresses(struct connman_network *network)
 	if (!service)
 		goto err;
 
-	connman_network_set_associating(network, false);
-
 	network->connecting = false;
 
 	ipconfig_ipv6 = __connman_service_get_ip6config(service);
@@ -511,6 +511,8 @@ static void check_dhcpv6(struct nd_router_advert *reply,
 	if (service) {
 		connman_service_create_ip6config(service, network->index);
 
+		connman_network_set_associating(network, false);
+
 		__connman_service_ipconfig_indicate_state(service,
 					CONNMAN_SERVICE_STATE_CONFIGURATION,
 					CONNMAN_IPCONFIG_TYPE_IPV6);
-- 
1.7.10.4

